packages/trader/src/AppV2/Components/OnboardingGuide/TradeParamsGuide/trade-params-guide.tsx

```diff
@@ -0,0 +1,50 @@
+import { useLocalStorageData } from '@deriv/hooks';
+import { Localize } from '@deriv/translations';
+import { getLocalStorage } from '@deriv/utils';
+import React from 'react';
+import { Step } from 'react-joyride';
+import GuideContainer from '../GuideForPages/guide-container';
+
+const TradeParamsGuide = () => {
+    const [guide_dtrader_v2, setGuideDtraderV2] = useLocalStorageData<Record<string, boolean>>('guide_dtrader_v2');
+    const [show_guide, setShowGuide] = React.useState(false);
+    const timerRef = React.useRef<NodeJS.Timeout>();
+
+    React.useEffect(() => {
```

Comment:
There are 3 components (this, QuickAdjGuide and ActiveSymbolsList) with similar logic: 
1. extracting flag from localStorage, creating state and ref for timer
2. in useEffect we are storing timer id, updating flags, states and in return -cleaning timer. 
Maybe it can be extracted in the hook?

Reply:
seems like an over engineer/abstraction